apidef/oas/server.go

```diff
@@ -265,3 +290,132 @@ func (dt *DetailedTracing) Fill(api apidef.APIDefinition) {
 func (dt *DetailedTracing) ExtractTo(api *apidef.APIDefinition) {
 	api.DetailedTracing = dt.Enabled
 }
+
+// Event holds information about individual event to be configured on the API.
+type Event struct {
+	Enabled bool            `json:"enabled" bson:"enabled"`
+	Type    tykevent.Event  `json:"type" bson:"type"`
+	Action  tykevent.Action `json:"action" bson:"action"`
+	ID      string          `json:"id,omitempty" bson:"id,omitempty"`
+
+	WebhookCore
+}
+
+// WebhookCore stores the core information about a webhook event.
+type WebhookCore struct {
+	Name         string            `json:"name,omitempty" bson:"name,omitempty"`
+	URL          string            `json:"url" bson:"url"`
+	Method       string            `json:"method" bson:"method"`
+	Timeout      int64             `json:"timeout" bson:"timeout"`
+	BodyTemplate string            `json:"bodyTemplate,omitempty" bson:"bodyTemplate,omitempty"`
+	Headers      map[string]string `json:"headers,omitempty" bson:"headers,omitempty"`
+}
+
+// ToConfMap converts WebhookCore to map[string]interface{}
+// with apidef.WebHookHandlerConf structure for classic API definition compatibility.
+func (c *WebhookCore) ToConfMap(enabled bool, id string) (map[string]interface{}, error) {
```

Comment:
If WebhookCode embeds Event, the parameters to this function are unnecessary. I'm still not sure the function is necessary either, as we have Decode() on apidef.WebHookHandlerConf.

Reply:
We need this since the classic API definition requires it as a `map[string]interface{}`, but directly marshalling it won't be helpful since the json keys are using `snake_case` in classic API def, and camel case in OAS def

Reply:
moved it to `GetWebhookConf` as a receiver function on `Event`, so that the same pattern can be used for other events in the future